test/functional/rpc_packages.py

```diff
@@ -330,31 +292,22 @@ def assert_equal_package_results(self, node, testmempoolaccept_result, submitpac
     def test_submit_child_with_parents(self, num_parents, partial_submit):
         node = self.nodes[0]
         peer = node.add_p2p_connection(P2PTxInvStore())
-        # Test a package with num_parents parents and 1 child transaction.
-        package_hex = []
+
         package_txns = []
-        values = []
-        scripts = []
         for _ in range(num_parents):
-            parent_coin = self.coins.pop()
-            value = parent_coin["amount"]
-            (tx, txhex, value, spk) = make_chain(node, self.address, self.privkeys, parent_coin["txid"], value)
-            package_hex.append(txhex)
-            package_txns.append(tx)
-            values.append(value)
-            scripts.append(spk)
+            parent_tx = self.wallet.create_self_transfer(fee=DEFAULT_FEE)
+            package_txns.append(parent_tx)
             if partial_submit and random.choice([True, False]):
-                node.sendrawtransaction(txhex)
-        child_hex = create_child_with_parents(node, self.address, self.privkeys, package_txns, values, scripts)
-        package_hex.append(child_hex)
-        package_txns.append(tx_from_hex(child_hex))
+                node.sendrawtransaction(parent_tx["hex"])
+        child_tx = self.wallet.create_self_transfer_multi(utxos_to_spend=[tx["new_utxo"] for tx in package_txns], fee_per_output=10000) #DEFAULT_FEE
+        package_txns.append(child_tx)
 
-        testmempoolaccept_result = node.testmempoolaccept(rawtxs=package_hex)
-        submitpackage_result = node.submitpackage(package=package_hex)
+        testmempoolaccept_result = node.testmempoolaccept(rawtxs=[tx["hex"] for tx in package_txns])
+        submitpackage_result = node.submitpackage(package=[tx["hex"] for tx in package_txns])
 
         # Check that each result is present, with the correct size and fees
-        for i in range(num_parents + 1):
-            tx = package_txns[i]
+        for package_tx in package_txns:
+            tx = package_tx['tx']
             wtxid = tx.getwtxid()
```

Comment:
`create_self_transfer` already returns `wtxid`, maybe it makes sense for `create_self_transfer_multi` to also return it.

Reply:
Done in df7e74b4e4d1b67b4b9d226ba00ffc21548d5677. Thanks for the review.